packages/langium/src/grammar/validation/validation-resources-collector.ts

```diff
@@ -0,0 +1,90 @@
+/******************************************************************************
+ * Copyright 2022 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { MultiMap } from '../../utils/collections';
+import { stream } from '../../utils/stream';
+import { LangiumDocuments } from '../../workspace/documents';
+import { AbstractElement, Action, Grammar, Interface, isAction, isAlternatives, isGroup, isUnorderedGroup, ParserRule, Type } from '../generated/ast';
+import { getActionType, getRuleType } from '../internal-grammar-util';
+import { AstResources, collectTypeResources, TypeResources } from '../type-system/type-collector/all-types';
+import { mergeInterfaces, mergeTypesAndInterfaces } from '../type-system/types-util';
+import { Property } from '../type-system/type-collector/types';
+import { TypeToValidationInfo, ValidationResources } from '../workspace/documents';
+
+export class LangiumGrammarValidationResourcesCollector {
+
+    collectValidationResources(grammar: Grammar, documents: LangiumDocuments): ValidationResources {
```

Comment:
The `LangiumDocuments` could be injected in the constructor instead of passing them as parameter.